esphome/components/growatt_solar/growatt_solar.cpp

```diff
@@ -0,0 +1,68 @@
+#include "growatt_solar.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace growatt_solar {
+
+static const char *const TAG = "growatt_solar";
+
+static const uint8_t MODBUS_CMD_READ_IN_REGISTERS = 0x04;
+static const uint8_t MODBUS_REGISTER_COUNT = 33;
+
+void GrowattSolar::update() { this->send(MODBUS_CMD_READ_IN_REGISTERS, 0, MODBUS_REGISTER_COUNT); }
+
+void GrowattSolar::on_modbus_data(const std::vector<uint8_t> &data) {
+  auto publish_1_reg_sensor_state = [&](sensor::Sensor *sensor, size_t i, float unit) -> void {
+    float value = encode_uint16(data[i * 2], data[i * 2 + 1]) * unit;
+    if (sensor != nullptr)
+      sensor->publish_state(value);
```

Comment:
This way it doesnt need to encode even if the sensor is not set. (Same for the next one too)
```suggestion
    if (sensor == nullptr)
      return;
    float value = encode_uint16(data[i * 2], data[i * 2 + 1]) * unit;
    sensor->publish_state(value);
```

Reply:
done!